server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -497,10 +504,23 @@ public synchronized IndexShard createShard(
                 }
             };
             Directory directory = directoryFactory.newDirectory(this.indexSettings, path);
+            Directory remoteDirectory = null;
+            if (this.indexSettings.isRemoteStoreEnabled()) {
+                try {
+                    Repository repository = repositoriesService.repository("dragon-stone");
```

Comment:
This repository name is hardcoded for now. It would be made configurable as a part of https://github.com/opensearch-project/OpenSearch/issues/3286